### ISSUE TITLE
key-spacing non working properly when having objects with nested functions

### ISSUE THREAD

**Issue by vekexasia** [eslint/eslint#5066]
Hello, considering the following code on the demo
```js
/*eslint-env es6*/
/*eslint quotes: [2, "single"]  */
/*eslint key-spacing: [2, {"align":"colon", "afterColon": true}] */
/*eslint no-unused-vars: 0 */
/*eslint one-var: 0 */
/*eslint arrow-body-style: 0 */

var b = {
 a   : () => 1,
 aaaa: () => 1
};

var c = {
 a   : () => {
   return {
     b   : 'a',
     cccc: 'a'
   };
 },
 aaaa: () => 1
}

var d = {
 a   : () => ({
   b   : 'a',
   cccc: 'a'
 }),
 aaaa: () => 1
}

var c1 = {
 aaaa: () => 1,
 a   : () => {
   return {
     b   : 'a',
     cccc: 'a'
   };
 }
}
```

Line 14,24 are complaining about "a" having an extra space after the key ('a'). But If i invert the `aaaa` with `a` then i get no complaints.

Is this the desired behavior?

**Comment by eslintbot** [comment]
@vekexasia Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by vekexasia** [comment]
Apparently the same behavior is also applied for the following code snippet:

```js
/*eslint-env es6*/
/*eslint quotes: [2, "single"]  */
/*eslint key-spacing: [2, {"align":"colon", "afterColon": true}] */
/*eslint no-unused-vars: 0 */
/*eslint one-var: 0 */
/*eslint arrow-body-style: 0 */


var c = {
 a   : {
   b   : 'a',
   cccc: 'a'
 },
 aaaa: () => 1
}

var c1 = {
 aaaa: () => 1,
 a   : {
   b   : 'a',
   cccc: 'a'
 } 
}
```

**Comment by nzakas** [comment]
Per the [comment](https://github.com/eslint/eslint/issues/5066#issuecomment-174566330), can you give the actual output from ESLint?

**Comment by vekexasia** [comment]
the error is: "Extra space after key "a". (key-spacing)"

**Comment by btmills** [comment]
This is a duplicate of #2668. Right now, alignment groups only continue across keys on adjacent lines. #2668 will make that behavior configurable. It's accepted and just waiting for an implementation - @vekexasia want to send a PR?

**Comment by vekexasia** [comment]
@btmills If only i had the knowledge to do that i would love to submit a pr. I'm afraid i won't be able to achieve anything useful. Maybe adding a reference to the code where you think the work should be done would kickstart someone who might be able/have the time to work on this.
